core/object/undo_redo.cpp

```diff
@@ -63,6 +63,23 @@ void UndoRedo::_discard_redo() {
 	actions.resize(current_action + 1);
 }
 
+void UndoRedo::cancel_action() {
+	ERR_FAIL_COND(action_level <= 0);
+	action_level--;
+
+	Action head_action = actions.write[actions.size() - 1];
```

Comment:
```suggestion
	Action head_action = actions[actions.size() - 1];
```
You don't need `write` here, unless instead you do:
```suggestion
	Action &head_action = actions.write[actions.size() - 1];
```
It is only for getting a mutable reference

Reply:
Thanks! I think I do need to get a mutable reference to delete the operations on the action. At least, that's what `_pop_history_tail()` does.

Reply:
Your code is copying it so I'd say to use the reference version, the second suggestion